packages/frontend/src/features/notification/types/NotificationTypes.tsx

```diff
@@ -0,0 +1,169 @@
+import { ReactComponent as UpvoteSuccessIcon } from '@/assets/svg/notification/upvote-success.svg'
+
+export type NotificationAction = {
+    label: string
+    type: string // Use this to identify actions dynamically
+}
+
+export interface NotificationConfig {
+    type: string
+    message: string
+    actions?: NotificationAction[]
+    icon?: React.FunctionComponent<React.SVGProps<SVGSVGElement>>
+}
+
+export const notificationConfig: Record<string, NotificationConfig> = {
+    SIGN_UP_SUCCESS: {
+        type: 'SIGN_UP_SUCCESS',
+        message:
+            '恭喜註冊成功，歡迎加入Unirep Social TW。有任何疑問可以參閱「平台說明」頁面',
+        icon: SignupSuccessIcon,
+    },
+    POST_SUCCEEDED: {
+        type: 'POST_SUCCEEDED',
+        message: '你的貼文發佈成功！',
+        actions: [{ label: '前往查看', type: 'viewPost' }],
+        icon: PostSuccessIcon,
+    },
+    POST_FAILED: {
+        type: 'POST_FAILED',
+        message: '你的貼文發佈未成功，可透過連結前往查看，並再嘗試發佈一次',
+        actions: [{ label: '前往查看', type: 'rewritePost' }],
+        icon: PostFailedIcon,
+    },
+    COMMENT_SUCCEEDED: {
+        type: 'COMMENT_SUCCEEDED',
+        message: '你的留言發佈成功！',
+        actions: [
+            // { label: "查看區塊鏈瀏覽器", execute: () => console.log("Navigating to blockchain explorer") },
+            { label: '前往查看', type: 'viewComment' },
+        ],
+        icon: CommentSuccessIcon,
+    },
+    COMMENT_FAILED: {
+        type: 'COMMENT_FAILED',
+        message: '你的留言發佈未成功，可透過連結前往查看，並再嘗試發佈一次',
+        actions: [{ label: '前往查看', type: 'rewriteComment' }],
+        icon: CommentFailedIcon,
+    },
+    UPVOTE_SUCCEEDED: {
+        type: 'UPVOTE_SUCCEEDED',
+        message: '你的貼文按讚成功！',
+        actions: [{ label: '前往查看', type: 'viewPost' }],
+        icon: UpvoteSuccessIcon,
+    },
+    UPVOTE_FAILED: {
+        type: 'UPVOTE_FAILED',
+        message: '你的貼文按讚未成功，可透過連結前往查看，並再嘗試按讚一次',
+        actions: [{ label: '前往查看', type: 'viewPost' }],
+        icon: UpvoteFailedIcon,
+    },
+    DOWNVOTE_SUCCEEDED: {
+        type: 'DOWNVOTE_SUCCEEDED',
+        message: '你的貼文倒讚成功！',
+        actions: [{ label: '前往查看', type: 'viewPost' }],
+        icon: DownvoteSuccessIcon,
+    },
+    DOWNVOTE_FAILED: {
+        type: 'DOWNVOTE_FAILED',
+        message: '你的貼文倒讚未成功，可透過連結前往查看，並再嘗試倒讚一次',
+        actions: [{ label: '前往查看', type: 'viewPost' }],
+        icon: DownvoteFailedIcon,
+    },
+    REPORT_SUCCEEDED: {
+        type: 'REPORT_SUCCEEDED',
+        message: '你的不當內容檢舉申請送出成功！待 5 位平台用戶進行評判',
+        icon: ReportSuccessIcon,
+    },
+    REPORT_FAILED: {
+        type: 'REPORT_FAILED',
+        message:
+            '你的不當內容檢舉申請送出未成功，可透過連結前往查看，並再嘗試送出申請一次',
+        actions: [{ label: '前往查看', type: 'reportDialog' }],
+        icon: ReportFailedIcon,
+    },
+    REPORT_PASSED: {
+        type: 'REPORT_PASSED',
+        message: '你的貼文檢舉案件被評判通過，你的聲譽分數提高 3 分',
+        actions: [
+            // { label: "查看區塊鏈瀏覽器", execute: () => console.log("Navigating to blockchain explorer") },
```

Comment:
I commented this line because i'm not sure if we need this feature, as the description in the github issue does not have this feature, while it does have it in the figma (but no further instruction for what it should do).